src/lib/block_producer/block_producer.ml

```diff
@@ -679,3 +679,240 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
         ignore
           (Time.Timeout.create time_controller time_till_genesis ~f:(fun _ ->
                start () )) )
+
+type precomputed_block =
```

Comment:
Would it be clearer if this new code were in its own module, like `Precomputed_block`?